src/coreclr/jit/emitxarch.cpp

```diff
@@ -6847,10 +6898,10 @@ void emitter::emitIns_R_R_A(
     id->idIns(ins);
     id->idReg1(reg1);
     id->idReg2(reg2);
-    if (instOptions == INS_OPTS_EVEX_b)
+    if (instOptions == INS_OPTS_EVEX_eb_er_rn)
     {
         assert(UseEvexEncoding());
-        id->idSetEvexbContext();
+        id->idSetEvexbContext(instOptions);
     }
```

Comment:
Should we assert that none of the other options like `INS_OPTS_EVEX_er_rd` are set in the `else` case?

Reply:
Same goes for other cases. Maybe we could have a general helper like `SetEvexBroadcastIfNeeded(id, instOptions)` and it can basically do:
```cpp
if (instOptions == INS_OPTS_EVEX_eb_er_rn)
{
    assert(UseEvexEncoding());
    id->idSetEvexbContext(instOptions);
}
else
{
    assert(instOptions == 0);
}
```

or similar...